schema/table.go

```diff
@@ -309,12 +309,13 @@ func (tt Tables) ValidateDuplicateColumns() error {
 }
 
 func (tt Tables) ValidateDuplicateTables() error {
+	tableNames := tt.TableNames()
```

Comment:
Previous code only validated top level tables. Fixed it so it uses `TableNames()` which gives us relations too, and added a test